sim/midas/src/main/cc/bridges/autocounter.cc

```diff
@@ -92,25 +93,49 @@ void autocounter_t::init() {
     write(mmio_addrs->init_done, 1);
 }
 
-template <typename T>
-void write_header_array_to_csv(std::ofstream& f, T* array, int length, std::string first_column) {
+std::string replace_all(std::string str, const std::string& from, const std::string& to) {
+    size_t start_pos = 0;
+    while((start_pos = str.find(from, start_pos)) != std::string::npos) {
+        str.replace(start_pos, from.length(), to);
+        start_pos += to.length();
+    }
+    return str;
+}
+
+std::string quote_csv_element(std::string str) {
+    std::string quoted = replace_all(str, "\"", "\"\"");
```

Comment:
Is this really what you want?  Is it going to replace double quotes in the description with two sets of double quotes?  Should it be adding an escape backslash before the double-quote in the description?

Reply:
In other words, I'm asking if you really mean 
```suggestion
    std::string quoted = replace_all(str, "\"", "\\\"");
```

I'm going to go try to run your Scala test for AutoCounterModule and look at the CSV that is generated to see if it makes sense.

This junk is why I was alright with you blowing up if someone passes a description that contains a comma.  It's unlikely to be a real issue and doesn't add the complexity of having to correctly handle quoting.

Reply:
I can't explain how happy it makes me that github allows you to to put poo emojis in code blocks...

```
// you should replace your code with
💩 💩 💩 
```
🤣

Reply:
My understanding is that you escape double quotes in csv by doubing them, not by using a backslash. 
https://stackoverflow.com/questions/17808511/properly-escape-a-double-quote-in-csv

Quotes shouldn't be escaped in the description in a csv-consistent way, since the description should not be assumed to be bound for CSV. Worst case, the user gets an extra pair of double quotes -- i think that's alright.

Reply:
I think that I want to run
```
testonly firesim.midsasexamples.AutoCounterModule
```

to run only the AutoCounterModule test.

Reply:
You want this: 

```testOnly firesim.midasexamples.AutoCounterModuleF1Test ```

You need to be in the firesim subproject to make that work though.

Reply:
> My understanding is that you escape double quotes in csv by doubing them, not by using a backslash.

Thanks for the pointer.  I hadn't seen that before.  It would be good to include a pointer to that SO question and then ignore my escaping with backslash.

I looked and `pandas.read_csv(doublequote=True)` (the default) follows this behavior as well:
![image](https://user-images.githubusercontent.com/59622590/152597454-3adb1aa0-057c-4ccb-88fc-0010e3626c0d.png)